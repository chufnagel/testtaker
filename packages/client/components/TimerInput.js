import PropTypes from "prop-types";

const TimerInput = ({ handleSettings, minutes }) => (
  <div className="timer-container" style={{ marginLeft: 100 }}>
    <h3>Set a time limit</h3>
    <input
      name="timer"
      value={minutes}
      type="number"
      minutes={minutes}
      onChange={handleSettings}
      required
    />
  </div>
);

// TimerInput.propTypes = {
//   minutes: PropTypes.string,
//   handleSettings: PropTypes.func.isRequired,
// };

TimerInput.defaultProps = {
  minutes: 20,
};

export default TimerInput;
