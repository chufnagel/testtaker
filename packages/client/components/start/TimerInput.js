import React, { Component } from "react";
import PropTypes from "prop-types";

const TimerInput = ({ action, minutes }) => (
  <div className="timer-container" style={{ marginLeft: 100 }}>
    <h3>Set a time limit</h3>
    <input
      min="0"
      minutes={minutes}
      name="minutes"
      type="number"
      onChange={action}
      value={minutes}
      required
    />
  </div>
);

TimerInput.propTypes = {
  // minutes: PropTypes.number,
  minutes: PropTypes.node,
  action: PropTypes.func.isRequired,
};

TimerInput.defaultProps = {
  minutes: 20,
};

export default TimerInput;
