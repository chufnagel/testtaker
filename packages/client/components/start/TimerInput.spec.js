/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import TimerInput from "./TimerInput";

// TimerInput.propTypes = {
//   action: PropTypes.func.isRequired,
//   path: PropTypes.string.isRequired,
//   query: PropTypes.string.isRequired,
// };

const mockFn = jest.fn();

describe("TimerInput", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TimerInput action={mockFn} />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("TimerInput matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
