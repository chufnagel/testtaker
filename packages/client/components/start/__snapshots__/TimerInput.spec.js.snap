// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerInput TimerInput matches snapshot 1`] = `
<div
  className="timer-container"
  style={
    Object {
      "marginLeft": 100,
    }
  }
>
  <h3>
    Set a time limit
  </h3>
  <input
    min="0"
    minutes={20}
    name="minutes"
    onChange={[MockFunction]}
    required={true}
    type="number"
    value={20}
  />
</div>
`;
